cloudinit/cmd/query.py

```diff
@@ -65,6 +65,21 @@ def get_parser(parser=None):
     return parser
 
 
+def maybe_decode_decompress_userdata(ud_file_path):
+    """Return a string of user-data from ud_file_path
+
+    Attempt to decode or decompress if needed
+    """
+    try:
+        return util.load_file(ud_file_path)
+    except UnicodeDecodeError:
+        gz_data = util.load_file(ud_file_path, decode=False)
+        try:
+            return util.decomp_gzip(gz_data)
+        except util.DecompressionError:
+            return gz_data
```

Comment:
I don't know if that may pose an issue, but returning the compressed data can cause an issue in a later stage for cloud-init ?

Reply:
Good question, the entire dictionary of processed meta-data and user-data is pushed through  util.json_dumps which has a default serialization function if it can't properly parse a valid json value via `json_serialize_default`.  So, if we pass through a valid that is not handled by our json_dumps, it gets base64encoded and we add the prefix "ci-b64:" to the value. I've added unit tests to cover these cases.